vcd/resource_vcd_vapp_static_routing.go

```diff
@@ -0,0 +1,211 @@
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"log"
+)
+
+func resourceVcdVappNetworkStaticRouting() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVappNetworkStaticRoutingCreate,
+		Delete: resourceVAppNetworkStaticRoutingDelete,
+		Read:   resourceVappNetworkStaticRoutingRead,
+		Update: resourceVappNetworkStaticRoutingUpdate,
+		Importer: &schema.ResourceImporter{
+			State: vappNetworkStaticRoutingImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"vapp_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "vApp identifier",
+			},
+			"network_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "vApp network identifier",
+			},
+			"enabled": &schema.Schema{
```

Comment:
Looking at the UI and `nat_enabled`, `firewall_enabled` it looks to me that static routing should also have the same toggle in `resource "vcd_vapp_network"` as opposed to resource itself for consitently handing enable/disable in all these resources (nat, firewall, static routing)


cc @dataclouder , @lvirbalas

Reply:
I have same idea, but choose todo is here because: fw and NAT applicable to vapp network and vapp org networks. Routing is only accessible for vapp network which is connected. Also in UI there is seperate Routing link and seperate toggle.
![image](https://user-images.githubusercontent.com/41415459/85658201-a749e980-b6bb-11ea-968e-afa0792cbde5.png)

Reply:
@vbauzysvmware I just realized comments/asks about:
> fw and NAT applicable to vapp network and vapp org networks

1.) In the FW page you have the below handy sentence. Should you a similar one to the NAT page?
> Firewall rules can be applied to networks connected to Org network or vApp networks which are fenced.

2.) In the NAT page example you enable the services:
```
firewall_enabled = true
nat_enabled      = true
```
But in the FW page enabling is missing.

3.) Apart from the above snippet in the example, we are not telling the user how to enable them neither in FW nor in NAT page. We need to do this. I suggest adding a sentence in the first paragraph. For example (I'm bolding the addition only here - no need in actual code):

> Provides a vCloud Director vApp Firewall resource. This can be used to create, modify, and delete firewall settings and rules in a vApp network. Firewall rules can be applied to networks connected to Org network or vApp networks which are fenced. **To enable the service use `firewall_enabled` argument in the network resource.**

Reply:
1.) The sentence is not precise in the NAT page:

<img width="696" alt="Screenshot 2020-06-25 at 12 28 57" src="https://user-images.githubusercontent.com/11330689/85694050-72e82480-b6df-11ea-8da5-b8559b595bad.png">

2.) These arguments should be code-quoted.

Reply:
done

Reply:
> Looking at the UI and nat_enabled, firewall_enabled it looks to me that static routing should also have the same toggle in resource "vcd_vapp_network" as opposed to resource itself for consitently handing enable/disable in all these resources (nat, firewall, static routing)

For me it actually looks the opposite, that it would be better to have the `enabled = true|false` flag **inside** the NAT and FW resources respectivel, just like we have here in Static Routing. Reasoning:
- Easier for the user to manage, because it is inside the same resource which actually contains the rules.
- Less jumping in the docs with links to other resources pointing the user how to enable/disable.
- Consistent with the API `types.go` where each of these three resources has ```IsEnabled        bool            `xml:"IsEnabled"``` inside itself.

Reply:
Another reason, I think this would remove these kinds of complexities from `vapp.go` in go-vcloud-director:
![Screenshot 2020-06-25 at 13 41 41](https://user-images.githubusercontent.com/11330689/85705345-cd867e00-b6e9-11ea-949d-a9c8a4fd6dc3.png)

Reply:
We decided to move `enabled` flag to FW and NAT resources as in Static Routing.

Reply:
Refactoring has been done, now we are polishing it in other comments.